core/node/btc_watch/src/lib.rs

```diff
@@ -0,0 +1,125 @@
+use std::time::Duration;
+
+use anyhow::Context as _;
+use tokio::sync::watch;
+use via_btc_client::{indexer::BitcoinInscriptionIndexer, types::BitcoinTxid};
+use zksync_dal::{Connection, ConnectionPool, Core, CoreDal};
+
+use crate::event_processors::EventProcessorError;
+
+mod event_processors;
+
+// re-exporting here isn't necessary, but it's a good practice to keep all the public types in one place
+pub use via_btc_client::types::BitcoinNetwork;
+
+#[derive(Debug)]
+struct BtcWatchState {
+    last_processed_bitcoin_block: u32,
+}
+
+#[derive(Debug)]
+pub struct BtcWatch {
+    indexer: BitcoinInscriptionIndexer,
+    poll_interval: Duration,
+    last_processed_bitcoin_block: u32,
+    pool: ConnectionPool<Core>,
+}
+
+impl BtcWatch {
+    pub async fn new(
+        rpc_url: &str,
+        network: BitcoinNetwork,
+        bootstrap_txids: Vec<BitcoinTxid>,
+        pool: ConnectionPool<Core>,
+        poll_interval: Duration,
+    ) -> anyhow::Result<Self> {
+        let indexer = BitcoinInscriptionIndexer::new(rpc_url, network, bootstrap_txids).await?;
+        let mut storage = pool.connection_tagged("btc_watch").await?;
+        let state = Self::initialize_state(&indexer, &mut storage).await?;
+        tracing::info!("initialized state: {state:?}");
+        drop(storage);
+
+        // TODO: init event processors
```

Comment:
I wouldn't call them the Event Processors, since there is no concept of the Event in the Bitcoin, maybe something like Inscription Processors.